holoviews/core/data/iris.py

```diff
@@ -232,7 +253,7 @@ def select_to_constraint(cls, selection):
             if isinstance(constraint, slice):
                 constraint = (constraint.start, constraint.stop)
             if isinstance(constraint, tuple):
-                constraint = iris.util.between(*constraint)
+                constraint = iris.util.between(*constraint, rh_inclusive=False)
```

Comment:
Ah, this must be related to the bounds issue I noticed in the tests (see below).